x-pack/plugins/observability/server/services/slo/slo_repository.ts

```diff
@@ -96,7 +96,7 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
 function buildFilterKuery(criteria: Criteria): string | undefined {
   const filters: string[] = [];
   if (!!criteria.name) {
-    filters.push(`slo.attributes.name: ${criteria.name}`);
+    filters.push(`slo.attributes.name: ${addWildcardIfAbsent(criteria.name)}`);
```

Comment:
💬 search on saved objects is using exact matches if no `*` is used. For example searching for "lat" yields only SLO with "lat" words, vs "lat*" yielding SLO like "latency"